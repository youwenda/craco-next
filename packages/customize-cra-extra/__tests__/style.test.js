const { override } = require('../lib');
const { addSassLoader } = require('../lib');
const { addCssLoader } = require('../lib');
const { addLessLoader } = require('../lib');
const { addPostcssPlugins } = require('../lib');

const { getConfig, traverse } = require('./config');

describe('addLessLoader', () => {
  test('addLessLoader', () => {
    const developmentConfig = getConfig();
    expect(
      traverse(addLessLoader()(developmentConfig)).module.rules[1].oneOf.slice(-3)
    ).toMatchSnapshot();
  });
  test('addLessLoader use less options', () => {
    const developmentConfig = getConfig();
    expect(
      traverse(
        addLessLoader({
          modifyVars: {
            'primary-color': '#4880FF',
            'link-color': '#4880FF',
            'font-size-base': '12px',
            'text-color': '#333',
            'text-color-secondary': '#999',
            'success-color': '#6AC500',
            'warning-color': '#FFA033',
            'heading-color': '#333',
            'border-color-base': '#E6E6E6',
            'primary': '#4D7FFF',
            'primary_hover': '#70a0ff',
            'green_2l': '70a0ff',
            'radius': '12px',
          },
          javascriptEnabled: true,
        })(developmentConfig)
      ).module.rules[1].oneOf.slice(-3)
    ).toMatchSnapshot();
  });
  test('addLessLoader use less other options', () => {
    const developmentConfig = getConfig();
    expect(
      traverse(
        addLessLoader({
          webpackImporter: false,
          lessOptions: {},
        })(developmentConfig)
      ).module.rules[1].oneOf.slice(-3)
    ).toMatchSnapshot();
  });
  test('addLessLoader use style-loader', () => {
    process.env.USE_STYLE_LOADER = 'true';
    const developmentConfig = getConfig();
    expect(
      traverse(addLessLoader()(developmentConfig)).module.rules[1].oneOf.slice(-3)
    ).toMatchSnapshot();
    delete process.env.USE_STYLE_LOADER;
  });
});

describe('addCssLoader', () => {
  test('default use mini-css-extract-plugin', () => {
    const developmentConfig = getConfig();
    expect(
      traverse(addCssLoader()(developmentConfig)).module.rules[1].oneOf.slice(5, 7)
    ).toMatchSnapshot();
  });
  test('use style-loader', () => {
    process.env.USE_STYLE_LOADER = 'true';
    const developmentConfig = getConfig();
    expect(
      traverse(addCssLoader()(developmentConfig)).module.rules[1].oneOf.slice(5, 7)
    ).toMatchSnapshot();
    delete process.env.USE_STYLE_LOADER;
  });
});

describe('style', () => {
  test('test style', () => {
    const developmentConfig = getConfig();
    expect(
      traverse(
        override(
          addLessLoader(),
          addSassLoader(),
          addCssLoader(),
          addPostcssPlugins([
            [
              'postcss-px-to-viewport',
              {
                unitToConvert: 'rpx',
                viewportWidth: 750,
                unitPrecision: 5,
                replace: false,
              },
            ],
            'postcss-plugin-px2rem',
            require('postcss-normalize')(),
          ])
        )(developmentConfig)
      ).module.rules[1].oneOf
    ).toMatchSnapshot();
  });
});
