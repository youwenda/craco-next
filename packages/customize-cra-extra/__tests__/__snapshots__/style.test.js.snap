// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addCssLoader default use mini-css-extract-plugin 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.css\\$/,
    "sideEffects": true,
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
]
`;

exports[`addCssLoader use style-loader 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.css\\$/,
    "sideEffects": true,
    "test": /\\\\\\.css\\$/,
    "use": Array [
      "craco-next/node_modules/style-loader/dist/cjs.js",
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      "craco-next/node_modules/style-loader/dist/cjs.js",
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
]
`;

exports[`addLessLoader addLessLoader 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.less\\$/,
    "sideEffects": true,
    "test": /\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "exclude": Array [
      /\\^\\$/,
      /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
      /\\\\\\.html\\$/,
      /\\\\\\.json\\$/,
    ],
    "type": "asset/resource",
  },
]
`;

exports[`addLessLoader addLessLoader use less options 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.less\\$/,
    "sideEffects": true,
    "test": /\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "lessOptions": Object {
            "javascriptEnabled": true,
            "modifyVars": Object {
              "border-color-base": "#E6E6E6",
              "font-size-base": "12px",
              "green_2l": "70a0ff",
              "heading-color": "#333",
              "link-color": "#4880FF",
              "primary": "#4D7FFF",
              "primary-color": "#4880FF",
              "primary_hover": "#70a0ff",
              "radius": "12px",
              "success-color": "#6AC500",
              "text-color": "#333",
              "text-color-secondary": "#999",
              "warning-color": "#FFA033",
            },
          },
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "lessOptions": Object {
            "javascriptEnabled": true,
            "modifyVars": Object {
              "border-color-base": "#E6E6E6",
              "font-size-base": "12px",
              "green_2l": "70a0ff",
              "heading-color": "#333",
              "link-color": "#4880FF",
              "primary": "#4D7FFF",
              "primary-color": "#4880FF",
              "primary_hover": "#70a0ff",
              "radius": "12px",
              "success-color": "#6AC500",
              "text-color": "#333",
              "text-color-secondary": "#999",
              "warning-color": "#FFA033",
            },
          },
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "exclude": Array [
      /\\^\\$/,
      /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
      /\\\\\\.html\\$/,
      /\\\\\\.json\\$/,
    ],
    "type": "asset/resource",
  },
]
`;

exports[`addLessLoader addLessLoader use less other options 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.less\\$/,
    "sideEffects": true,
    "test": /\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "lessOptions": Object {},
          "sourceMap": true,
          "webpackImporter": false,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "lessOptions": Object {},
          "sourceMap": true,
          "webpackImporter": false,
        },
      },
    ],
  },
  Object {
    "exclude": Array [
      /\\^\\$/,
      /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
      /\\\\\\.html\\$/,
      /\\\\\\.json\\$/,
    ],
    "type": "asset/resource",
  },
]
`;

exports[`addLessLoader addLessLoader use style-loader 1`] = `
Array [
  Object {
    "exclude": /\\\\\\.module\\\\\\.less\\$/,
    "sideEffects": true,
    "test": /\\\\\\.less\\$/,
    "use": Array [
      "craco-next/node_modules/style-loader/dist/cjs.js",
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.less\\$/,
    "use": Array [
      "craco-next/node_modules/style-loader/dist/cjs.js",
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "exclude": Array [
      /\\^\\$/,
      /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
      /\\\\\\.html\\$/,
      /\\\\\\.json\\$/,
    ],
    "type": "asset/resource",
  },
]
`;

exports[`style test style 1`] = `
Array [
  Object {
    "mimetype": "image/avif",
    "parser": Object {
      "dataUrlCondition": Object {
        "maxSize": 10000,
      },
    },
    "test": Array [
      /\\\\\\.avif\\$/,
    ],
    "type": "asset",
  },
  Object {
    "parser": Object {
      "dataUrlCondition": Object {
        "maxSize": 10000,
      },
    },
    "test": Array [
      /\\\\\\.bmp\\$/,
      /\\\\\\.gif\\$/,
      /\\\\\\.jpe\\?g\\$/,
      /\\\\\\.png\\$/,
    ],
    "type": "asset",
  },
  Object {
    "issuer": Object {
      "and": Array [
        /\\\\\\.\\(ts\\|tsx\\|js\\|jsx\\|md\\|mdx\\)\\$/,
      ],
    },
    "test": /\\\\\\.svg\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/@svgr/webpack/lib/index.js",
        "options": Object {
          "prettier": false,
          "ref": true,
          "svgo": false,
          "svgoConfig": Object {
            "plugins": Array [
              Object {
                "removeViewBox": false,
              },
            ],
          },
          "titleProp": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/file-loader/dist/cjs.js",
        "options": Object {
          "name": "static/media/[name].[hash].[ext]",
        },
      },
    ],
  },
  Object {
    "include": "craco-next/packages/customize-cra-extra/src",
    "loader": "craco-next/node_modules/babel-loader/lib/index.js",
    "options": Object {
      "babelrc": false,
      "cacheCompression": false,
      "cacheDirectory": true,
      "cacheIdentifier": "development:babel-plugin-named-asset-import@0.3.8:babel-preset-react-app@10.0.1:react-dev-utils@12.0.1:react-scripts@5.0.1",
      "compact": false,
      "configFile": false,
      "customize": "craco-next/node_modules/babel-preset-react-app/webpack-overrides.js",
      "plugins": Array [
        "craco-next/node_modules/react-refresh/babel.js",
      ],
      "presets": Array [
        Array [
          "craco-next/node_modules/babel-preset-react-app/index.js",
          Object {
            "runtime": "classic",
          },
        ],
      ],
    },
    "test": /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
  },
  Object {
    "exclude": /@babel\\(\\?:\\\\/\\|\\\\\\\\\\{1,2\\}\\)runtime/,
    "loader": "craco-next/node_modules/babel-loader/lib/index.js",
    "options": Object {
      "babelrc": false,
      "cacheCompression": false,
      "cacheDirectory": true,
      "cacheIdentifier": "development:babel-plugin-named-asset-import@0.3.8:babel-preset-react-app@10.0.1:react-dev-utils@12.0.1:react-scripts@5.0.1",
      "compact": false,
      "configFile": false,
      "inputSourceMap": true,
      "presets": Array [
        Array [
          "craco-next/node_modules/babel-preset-react-app/dependencies.js",
          Object {
            "helpers": true,
          },
        ],
      ],
      "sourceMaps": true,
    },
    "test": /\\\\\\.\\(js\\|mjs\\)\\$/,
  },
  Object {
    "exclude": /\\\\\\.module\\\\\\.css\\$/,
    "sideEffects": true,
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 1,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": true,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
    ],
  },
  Object {
    "exclude": /\\\\\\.module\\\\\\.less\\$/,
    "sideEffects": true,
    "test": /\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/less-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "exclude": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
    "sideEffects": true,
    "test": /\\\\\\.\\(scss\\|sass\\)\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "mode": "icss",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/sass-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.module\\\\\\.\\(scss\\|sass\\)\\$/,
    "use": Array [
      Object {
        "loader": "craco-next/node_modules/mini-css-extract-plugin/dist/loader.js",
        "options": Object {},
      },
      Object {
        "loader": "craco-next/node_modules/css-loader/dist/cjs.js",
        "options": Object {
          "importLoaders": 3,
          "modules": Object {
            "getLocalIdent": [Function],
            "mode": "local",
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/postcss-loader/dist/cjs.js",
        "options": Object {
          "postcssOptions": Object {
            "config": false,
            "ident": "postcss",
            "plugins": Array [
              "postcss-flexbugs-fixes",
              Array [
                "postcss-preset-env",
                Object {
                  "autoprefixer": Object {
                    "flexbox": "no-2009",
                  },
                  "stage": 4,
                },
              ],
              "postcss-normalize",
              Array [
                "postcss-px-to-viewport",
                Object {
                  "replace": false,
                  "unitPrecision": 5,
                  "unitToConvert": "rpx",
                  "viewportWidth": 750,
                },
              ],
              "postcss-plugin-px2rem",
              Object {
                "Once": [Function],
                "postcssImport": [Function],
                "postcssPlugin": "postcss-normalize",
              },
            ],
          },
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/resolve-url-loader/index.js",
        "options": Object {
          "root": "craco-next/packages/customize-cra-extra/src",
          "sourceMap": false,
        },
      },
      Object {
        "loader": "craco-next/node_modules/sass-loader/dist/cjs.js",
        "options": Object {
          "sourceMap": true,
        },
      },
    ],
  },
  Object {
    "exclude": Array [
      /\\^\\$/,
      /\\\\\\.\\(js\\|mjs\\|jsx\\|ts\\|tsx\\)\\$/,
      /\\\\\\.html\\$/,
      /\\\\\\.json\\$/,
    ],
    "type": "asset/resource",
  },
]
`;
